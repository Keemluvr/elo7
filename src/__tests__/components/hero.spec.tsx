import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Hero from "@/components/hero";

describe("Hero component", () => {
  beforeEach(() => {
    render(<Hero />);
  });

  test("should render the section component", () => {
    const sectionElement = screen.getByRole("region", {
      name: "seção de banner",
    });

    expect(sectionElement).toBeInTheDocument();
  });

  test("should render the image with correct properties", () => {
    const imageElement = screen.getByRole("img", { name: "imagem de fundo" });
    const expectedSrc = "/_next/image?url=%2Fimages%2Fhero.webp&w=3840&q=100";

    expect(imageElement).toBeInTheDocument();
    expect(imageElement).toHaveClass("hero-image");
    expect(imageElement).toHaveAttribute("src", expectedSrc);
    expect(imageElement).toHaveAttribute("fetchpriority", "high");
    expect(imageElement).toHaveAttribute("srcset");
    expect(imageElement.getAttribute("srcset")).not.toBe("");
  });

  test("should render the title with correct properties", () => {
    const titleElement = screen.getByRole("heading", {
      name: "titulo emcima da imagem",
    });

    expect(titleElement).toBeInTheDocument();
    expect(titleElement).toHaveClass("hero-image-title");
    expect(titleElement).toHaveTextContent("trabalhe no elo7");
    expect(titleElement.tagName).toBe("H1");
  });

  test("should render the description with correct properties", () => {
    const descriptionElement = screen.getByRole("paragraph", {
      name: "descrição da elo7",
    });

    expect(descriptionElement).toBeInTheDocument();
    expect(descriptionElement).toHaveClass("hero-description");
    expect(descriptionElement).toHaveTextContent(
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur et viverra orci. Praesent consequat dolor tellus, eget viverra risus hendrerit non. Sed rutrum condimentum maximus. Donec pellentesque libero eu eros sagittis."
    );
  });

  test("should render the horizontal line with correct properties", () => {
    const horizontalLine = screen.getByRole("separator", {
      name: "linha horizontal",
    });

    expect(horizontalLine).toBeInTheDocument();
    expect(horizontalLine).toHaveClass("hero-horizontal-line");
  });

  test("should render the link button with correct properties", () => {
    const linkButton = screen.getByRole("button", { name: "vagas em aberto" });

    expect(linkButton).toBeInTheDocument();
    expect(linkButton).toHaveAttribute("href", "#open-positions");
    expect(linkButton).toHaveClass("hero-button");
    expect(linkButton).toHaveRole("button");
    expect(linkButton.tagName).toBe("A");
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Hero />);
    expect(asFragment()).toMatchSnapshot();
  });
});
