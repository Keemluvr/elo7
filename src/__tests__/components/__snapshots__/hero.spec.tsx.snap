// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component matches snapshot 1`] = `
<DocumentFragment>
  <section
    aria-label="seção de banner"
  >
    <div
      class="hero-image-wrapper"
    >
      <img
        alt="Equipe de colaboradores celebrando e batendo as mãos em um ambiente de escritório, destacando a cultura de trabalho colaborativa e positiva da empresa."
        aria-label="imagem de fundo"
        class="hero-image"
        data-nimg="fill"
        decoding="async"
        fetchpriority="high"
        sizes="100vw"
        src="/_next/image?url=%2Fimages%2Fhero.webp&w=3840&q=100"
        srcset="/_next/image?url=%2Fimages%2Fhero.webp&w=640&q=100 640w, /_next/image?url=%2Fimages%2Fhero.webp&w=750&q=100 750w, /_next/image?url=%2Fimages%2Fhero.webp&w=828&q=100 828w, /_next/image?url=%2Fimages%2Fhero.webp&w=1080&q=100 1080w, /_next/image?url=%2Fimages%2Fhero.webp&w=1200&q=100 1200w, /_next/image?url=%2Fimages%2Fhero.webp&w=1920&q=100 1920w, /_next/image?url=%2Fimages%2Fhero.webp&w=2048&q=100 2048w, /_next/image?url=%2Fimages%2Fhero.webp&w=3840&q=100 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
      <h1
        aria-label="titulo emcima da imagem"
        class="hero-image-title"
      >
        trabalhe no elo7
      </h1>
    </div>
    <div
      class="grid-container"
    >
      <p
        aria-label="descrição da elo7"
        class="grid-item-full-width hero-description"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur et viverra orci. Praesent consequat dolor tellus, eget viverra risus hendrerit non. Sed rutrum condimentum maximus. Donec pellentesque libero eu eros sagittis.
      </p>
      <hr
        aria-label="linha horizontal"
        class="grid-item-full-width hero-horizontal-line"
      />
      <a
        aria-label="vagas em aberto"
        class="grid-item-full-width hero-button"
        href="#open-positions"
        role="button"
      >
        vagas em aberto
      </a>
    </div>
  </section>
</DocumentFragment>
`;
